src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -58,32 +55,28 @@ public sealed class ODataBatchReader : IODataBatchOperationListener
         /// <summary>
         /// Internal switch for whether we support reading Content-ID header appear in HTTP head instead of ChangeSet head.
         /// </summary>
-        private bool allowLegacyContentIdBehaviour;
+        private bool allowLegacyContentIdBehavior;
 
         /// <summary>
         /// Constructor.
         /// </summary>
         /// <param name="inputContext">The input context to read the content from.</param>
-        /// <param name="batchBoundary">The boundary string for the batch structure itself.</param>
-        /// <param name="batchEncoding">The encoding to use to read from the batch stream.</param>
         /// <param name="synchronous">true if the reader is created for synchronous operation; false for asynchronous.</param>
-        internal ODataBatchReader(ODataRawInputContext inputContext, string batchBoundary, Encoding batchEncoding, bool synchronous)
+        protected ODataBatchReader(ODataInputContext inputContext, bool synchronous)
         {
             Debug.Assert(inputContext != null, "inputContext != null");
-            Debug.Assert(!string.IsNullOrEmpty(batchBoundary), "!string.IsNullOrEmpty(batchBoundary)");
 
             this.inputContext = inputContext;
             this.container = inputContext.Container;
             this.synchronous = synchronous;
             this.payloadUriConverter = new ODataBatchPayloadUriConverter(inputContext.PayloadUriConverter);
-            this.batchStream = new ODataBatchReaderStream(inputContext, batchBoundary, batchEncoding);
-            this.allowLegacyContentIdBehaviour = true;
+            this.allowLegacyContentIdBehavior = true;

```

Comment:
>allowLegacyContentIdBehavior [](start = 17, length = 28)

It looks like this is always set to true, with no way to set to false.  Can we just get rid of the flag?

Reply:
It appears so from the code in master branch. Should we log a separate issue to investigate since it is not related to porting?

Reply:
It appears so from the code in master branch. Should we log a separate issue to investigate since it is not related to porting?
Let me know if you want me to log an issue in github, so that it can be looked into in multiple branches and fix it altogether.

---
In reply to: [140609951](https://github.com/OData/odata.net/pull/960#discussion_r140609951) [](ancestors = 140609951)